frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/workspace_overview/overview_block_component.dart

```diff
@@ -0,0 +1,326 @@
+        child = BlockComponentActionWrapper(
+          node: widget.node,
+          actionBuilder: widget.actionBuilder!,
+          child: child,
+        );
+      }
+    } else {
+      child = MobileBlockActionButtons(
+        node: node,
+        editorState: editorState,
+        child: child,
+      );
+    }
+
+    return child;
+  }
+
+  List<Widget> _buildOverviewBlockChildren(
+    ViewPB view,
+    TextDirection textDirection,
+    double leftIndent,
+  ) {
+    final children = <Widget>[];
+
+    children.add(
+      _buildOverviewItemWidget(
+        view,
+        textDirection,
+        leftIndent,
+      ),
+    );
+
+    for (final child in view.childViews) {
+      children.addAll(
+        _buildOverviewBlockChildren(
+          child,
+          textDirection,
+          leftIndent + leftIndentIncrementValue,
+        ),
+      );
+    }
+
+    return children;
+  }
+
+  Widget _buildOverviewItemWidget(
+    ViewPB view,
+    TextDirection textDirection,
+    double leftIndent,
+  ) {
+    return Container(
+      padding: const EdgeInsets.only(bottom: 5.0),
+      width: double.infinity,
+      child: OverviewItemWidget(
+        id: view.id,
+        text: view.name,
+        textDirection: textDirection,
+        leftIndent: leftIndent,
+        icon: view.icon,
+        defaultIcon: view.defaultIcon(),
+      ),
+    );
+  }
+}
+
+class OverviewItemWidget extends StatelessWidget {
+  const OverviewItemWidget({
+    super.key,
+    required this.id,
+    required this.text,
+    required this.textDirection,
+    required this.defaultIcon,
+    this.leftIndent = 0.0,
+    this.icon,
+  });
+
+  final String id;
+  final String text;
+  final ViewIconPB? icon;
+  final Widget defaultIcon;
+  final TextDirection textDirection;
+  final double leftIndent;
+
+  @override
+  Widget build(BuildContext context) {
+    final editorState = context.read<EditorState>();
+    final textStyle = editorState.editorStyle.textStyleConfiguration;
+    final style = textStyle.href.combine(textStyle.text);
+    return FlowyHover(
+      style: HoverStyle(hoverColor: Theme.of(context).hoverColor),
+      builder: (context, onHover) {
+        return GestureDetector(
+          behavior: HitTestBehavior.translucent,
+          onTap: _openPage,
+          child: Row(
+            textDirection: textDirection,
+            children: [
+              HSpace(leftIndent),
+              icon?.value.isNotEmpty ?? false
+                  ? EmojiText(emoji: icon!.value, fontSize: 18.0)
+                  : SizedBox.square(dimension: 20.0, child: defaultIcon),
+              const HSpace(8.0),
+              Text(
+                text,
+                style: style.copyWith(
+                  color: onHover
+                      ? Theme.of(context).colorScheme.onSecondary
+                      : null,
+                ),
+              ),
+            ],
+          ),
+        );
+      },
+    );
+  }
+
+  void _openPage() async {
```

Comment:
Async methods should return a `Future`
```suggestion
  Future<void> _openPage() async {
```